### ISSUE TITLE
Algorithm: Scan

### ISSUE THREAD

**Issue by ax3l** [ComputationalRadiationPhysics/pyDive#3]
Idea by @psychocoderHPC 

An algorithm similar to [MPI_Scan](http://www.mpich.org/static/docs/v3.1/www3/MPI_Scan.html) might be useful. Can this already be performed efficiently with the current `map`/`reduce`/`mapReduce` algorithms or do we have to add it?

**Comment by heikoburau** [comment]
This is a feature we need to add in a new algorithm. Any idea how to parallelize this?

**Comment by ax3l** [comment]
The [mpi standard](http://mpi-forum.org/docs/mpi-3.0/mpi30-report.pdf) defines _inclusive_ and _exclusive_ scan operations in section 5.11 that I am aware of.

Section 5.2.2 also describes the difference to other _collective_ operations (`scan` is neither all2all, all2one nor one2all). But I would also have to read the papers on how to parallelize that.

@psychocoderHPC can you describe your exact idea? :)

**Comment by heikoburau** [comment]
Can this issue be closed now?

**Comment by ax3l** [comment]
Is it implemented?

**Comment by heikoburau** [comment]
No.

**Comment by ax3l** [comment]
Well I think it is a very useful future nice-to-have don't you think? You can just create some milestones and archive it.

**Comment by heikoburau** [comment]
Yeah, I just thought it was merely a question, because of the question label.

**Comment by ax3l** [comment]
Good point, _transformed_ :tractor:
